apiary.apib

```diff
@@ -1706,22 +1706,802 @@ root will not be listable, as it is virtual. Access known paths directly.
 ## bw
 
 # Group swarm
+Swarm inspection tool.
 
-## addrs
+This command can't be called directly.
 
-### `swarm addrs local`
+## addrs [GET /swarm/addrs]
+List known addresses. Useful for debugging.
 
-## connect
+`ipfs swarm addrs` lists all addresses this node is aware of.
 
-## disconnect
++ Request Without Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/swarm/addrs"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/swarm/addrs"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: application/json
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 11 Feb 2016 23:00:03 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (object)
+        + Addrs (SwarmAddrs)
+
+    + Body
+
+        ```
+        {
+          "Addrs": {
+            "QmNfCubGpwYZAQxX8LQDsYgB48C4GbfZHuYdexpX9mbNyT": [
+              "/ip4/127.0.0.1/tcp/4001",
+              "/ip4/192.168.0.101/tcp/4001",
+              "/ip4/70.81.176.29/tcp/48695",
+              "/ip6/::1/tcp/4001"
+            ],
+            ...
+          }
+        }
+        ```
+
+## addrs local [GET /swarm/addrs/local{&id}]
+List all local addresses.
+
+`ipfs swarm addrs local` lists all local addresses the node is listening on.
+
++ Parameters
+    + id (boolean, optional) - Show peer ID in addresses
+
++ Request Without Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/swarm/addrs/local"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/swarm/addrs/local"
```

Comment:
Seems confusing to me, will take a look if there is a cleaner way to do this. But this can be fixed later so don't let it hold you up.